dagmc_stats.py

```diff
@@ -39,15 +39,117 @@ def get_entity_ranges(my_core, meshset, entity_types):
+            entity_ranges['Curves'] = my_core.get_entities_by_type_and_tag(meshset, types.MBENTITYSET, dagmc_tags['geom_dim'], [1])
     return entity_ranges
 
+
+def get_surfaces_per_volume(my_core, entity_ranges):
+    """
+    Get the number of surfaces that each volume in a given file contains
+    
+    inputs
+    ------
+    my_core : a MOAB core instance
+    entity_ranges : a dictionary of the ranges of each tag in a file
+    
+    outputs
+    -------
+    pdf : a histogram that displays the number of surfaces each volume has (matplotlib)
+    """
+    freqs = []
+    for volumeset in entity_ranges['Volumes']:
+        freqs.append(my_core.get_child_meshsets(volumeset).size())
+    stats = {}
+    stats['minimum'] = min(freqs)
+    stats['maximum'] = max(freqs)
+    stats['median'] = np.median(freqs)
+    stats['average'] = np.average(freqs)
+    return stats
+
+
+def make_verts_dictionary(vert_list):
+    divisions = int(len(vert_list)/10000)
+    broken_up_verts = []
+    for cut in range(divisions+1):
+        broken_up_verts.append(list(vert_list[cut*10000:cut*10000+10000]))
+    
+    vertices_and_frequencies = {}
+    for cut in range(divisions+1):
+        for vert in broken_up_verts[cut]:
+            number_of_vertex = vert
+            try:
+                vertices_and_frequencies[number_of_vertex] += 1
+            except KeyError:
+                vertices_and_frequencies[number_of_vertex] = 1
+    number_of_frequencies = {}
+    for vertex in vertices_and_frequencies:
+        try:
+            number_of_frequencies[vertices_and_frequencies[vertex]] += 1
+        except KeyError:
+            number_of_frequencies[vertices_and_frequencies[vertex]] = 1
+    sorted_dict = {}
+    for frequency in sorted(number_of_frequencies.keys()):
+        sorted_dict[frequency] = number_of_frequencies[frequency]
+    return(sorted_dict)
+
+
+def find_median(sorted_dict):
+    length = 0
+    for value in sorted_dict.values():
+        length += value
+    half = length / 2
+    sum_var = 0
+    for val in sorted_dict.values():
+        if half-sum_var > 0:
+            sum_var += val
+        else:
+            break
+    index = (list(sorted_dict.values()).index(val))
+    if sum(list(sorted_dict.values())[index:]) != sum(list(sorted_dict.values())[:index]):
+        if sum(list(sorted_dict.values())[index:]) > sum(list(sorted_dict.values())[:index]):
+            median = list(sorted_dict.keys())[index]
+        else:
+            median = list(sorted_dict.keys())[index-1]
+    else:
+        median = (list(sorted_dict.keys())[index-1] + list(sorted_dict.keys())[index]) / 2
+    return(median)
+        
+def find_mean(sorted_dict):
+    total = 0
+    for key, value in sorted_dict.items():
+        total += key*value
+    mean = total / sum(sorted_dict.values())
+    return(mean)
+
+def get_triangles_per_vertex(my_core, all_meshset):
```

Comment:
We should discuss this method in person.  It seems much more complex than necessary, perhaps based on your lack of familiarity with the MOAB database.

In short, I would implement this in a way that is algorithmically identical to your `get_surfaces_per_volume` method:

pseudo-code:
```
For each vertex in the while problem
       Query the number of triangles adjacent to this vertex
       Append this to a list (like `freqs`)

Perform some statistics: min, mean, median, max
```